search6/src/main/java/org/hibernate/performance/search/model/application/ModelServiceImpl.java

```diff
@@ -0,0 +1,127 @@
+			config.put( "hibernate.search.backend.directory.type", "local-heap" );
+		}
+
+		if ( kind.isManual() ) {
+			config.put( HibernateOrmMapperSettings.AUTOMATIC_INDEXING_STRATEGY, AutomaticIndexingStrategyName.NONE );
+		}
+
+		return config;
+	}
+
+	@Override
+	public <E> List<E> search(Session session, Class<E> entityClass, Integer limit) {
+		return Search.session( session ).search( entityClass ).where( f -> f.matchAll() ).fetchHits( limit );
+	}
+
+	@Override
+	public <E> List<E> search(Session session, Class<E> entityClass, String fieldName, Object value) {
+		return Search.session( session ).search( entityClass ).where(
+				f -> f.match().field( fieldName ).matching( value ) ).fetchHits( DEFAULT_LIMIT );
+	}
+
+	@Override
+	public <E> List<E> searchAnd(Session session, Class<E> entityClass, String fieldName1, Object value1,
+			String fieldName2, Object value2) {
+		SearchSession searchSession = Search.session( session );
+
+		SearchScope<E> scope = searchSession.scope( entityClass );
+		MatchPredicateOptionsStep<?> matching1 = scope.predicate().match().field( fieldName1 ).matching( value1 );
+		MatchPredicateOptionsStep<?> matching2 = scope.predicate().match().field( fieldName2 ).matching( value2 );
+		SearchPredicate predicate = scope.predicate().bool().must( matching1 ).must( matching2 ).toPredicate();
+
+		return searchSession.search( scope ).where( predicate ).fetchHits( DEFAULT_LIMIT );
+	}
+
+	@Override
+	public long count(Session session, Class<?> entityClass, String fieldName, Object value) {
+		return Search.session( session ).search( entityClass ).where(
+				f -> f.match().field( fieldName ).matching( value ) ).fetchTotalHitCount();
+	}
+
+	@Override
+	public <E> List<E> range(Session session, Class<E> entityClass, String fieldName, Object start, Object end) {
+		return Search.session( session ).search( entityClass ).where(
+				f -> f.range().field( fieldName )
+						// include limits
+						.between( start, RangeBoundInclusion.INCLUDED, end, RangeBoundInclusion.INCLUDED ) )
+				.fetchHits( DEFAULT_LIMIT );
+	}
+
+	@Override
+	public <E> List<E> rangeOrderBy(Session session, Class<E> entityClass, String fieldName, Object start, Object end) {
+		return Search.session( session ).search( entityClass ).where(
+				f -> f.range().field( fieldName )
+						// include limits
+						.between( start, RangeBoundInclusion.INCLUDED, end, RangeBoundInclusion.INCLUDED ) )
+				// sorted by the same field on which we apply the range
+				.sort( f -> f.field( fieldName ) )
+				.fetchHits( DEFAULT_LIMIT );
+	}
+
+	@Override
+	public List<Object> projectId(Session session, Class<?> entityClass, String fieldName, Object value) {
+		List<EntityReference> entityReferences = Search.session( session ).search( entityClass )
+				.selectEntityReference()
+				.where( f -> f.match().field( fieldName ).matching( value ) )
+				.fetchHits( DEFAULT_LIMIT );
+
+		return entityReferences.stream().map( a -> a.id() ).collect( Collectors.toList() );
```

Comment:
To avoid creating multiple lists:

```suggestion
		return Search.session( session ).search( entityClass )
				.select( f -> f.composite( ref -> ref.id(), f.entityReference() ) )
				.where( f -> f.match().field( fieldName ).matching( value ) )
				.fetchHits( DEFAULT_LIMIT );
```